utils/cache_manager.py

```
@@ -1,20 +1,131 @@
+            if generated_checksum != stored_checksum:
+                raise ValueError("Cache integrity check failed.")
+
+            # Reconstruct Flask.Response if applicable
+            if isinstance(value, dict) and "response_data" in value:
+                return Response(
+                    response=value["response_data"],
+                    status=value["status"],
+                    headers=value["headers"],
+                )
+            return value
+        except:
```

Ruff (0.8.0)

Do not use bare `except`

(E722)